x-pack/plugins/security_solution/public/detections/components/alerts_kpis/common/hooks.ts

```diff
@@ -48,3 +53,27 @@ export const useInspectButton = ({
     };
   }, [setQuery, loading, response, request, refetch, uniqueQueryId, deleteQuery]);
 };
+
+export const useStackByFields = () => {
+  const [stackByFieldOptions, setStackByFieldOptions] = useState<
+    undefined | EuiComboBoxOptionOption[]
+  >(undefined);
+  const { pathname } = useLocation();
+
+  const { browserFields } = useSourcererDataView(getScopeFromPath(pathname));
```

Comment:
Does `useSourcererDataView()` update during runtime sometimes or can we assume `browserFields` is static at this point?

Reply:
yes I believe it can be changed by a form control present on most pages that allows a user to change the index patter...data views that are being used to generate the data on a page, this hook will be re-evaluated then.

Reply:
Thanks :)